File: LeetCode-2023/1653.使字符串平衡的最少删除次数.py
#
# @lc app=leetcode.cn id=1653 lang=python3
#
# [1653] 使字符串平衡的最少删除次数
#

# @lc code=start
class Solution:
    def minimumDeletions(self, s: str) -> int:
        leftb = 0
        righta = s.count('a')
        result = righta
        for c in s:
            if c == 'a':
                righta -= 1
            else:
                leftb += 1
            result = min(result, leftb + righta)
        return result
# @lc code=end

File: LeetCode-2023/1653.使字符串平衡的最少删除次数.cpp
/*
 * @lc app=leetcode.cn id=1653 lang=cpp
 *
 * [1653] 使字符串平衡的最少删除次数
 */

// @lc code=start
class Solution {
public:
    int minimumDeletions(string s) {
        int leftb = 0, righta = 0;
        for(int i = 0; i < s.size(); i++) {
            if(s[i] == 'a') {
                righta++;
            }
        }
        int result = righta;
        for(int i = 0; i < s.size(); i++) {
            if(s[i] == 'a') {
                righta--;
            } else {
                leftb++;
            }
            result = min(result, leftb + righta);
        }
        return result;
    }
};
// @lc code=end

